docs/upgrade-guide-1.6-to-1.7.ipynb

```diff
@@ -191,7 +282,164 @@
+    "juju refresh katib-ui --channel 0.15/stable\n",
+    "juju refresh kfp-api --channel 2.0/stable\n",
+    "juju refresh kfp-db --channel latest/stable\n",
+    "juju refresh kfp-persistence --channel 2.0/stable\n",
+    "juju refresh kfp-profile-controller --channel 2.0/stable\n",
+    "juju refresh kfp-schedwf --channel 2.0/stable\n",
+    "juju refresh kfp-ui --channel 2.0/stable\n",
+    "juju refresh kfp-viewer --channel 2.0/stable\n",
+    "juju refresh kfp-viz --channel 2.0/stable\n",
+    "juju refresh kubeflow-dashboard --channel 1.7/stable\n",
+    "juju refresh kubeflow-profiles --channel 1.7/stable\n",
+    "juju refresh kubeflow-roles --channel 1.7/stable\n",
+    "juju refresh kubeflow-volumes --channel 1.7/stable\n",
+    "juju refresh metacontroller-operator --channel 2.0/stable\n",
+    "juju refresh minio --channel ckf-1.7/stable\n",
+    "juju refresh oidc-gatekeeper --channel ckf-1.7/stable\n",
+    "juju refresh seldon-controller-manager --channel 1.15/stable\n",
+    "juju refresh tensorboard-controller --channel 1.7/stable\n",
+    "juju refresh tensorboards-web-app --channel 1.7/stable\n",
+    "juju refresh training-operator --channel 1.6/stable"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "text"
+    ]
+   },
+   "source": [
+    "*Troubleshooting charm upgrade*\n",
+    "\n",
+    "If charm fails upgrade or is stuck in `maintenance` state for long time it is possible to recover by running refresh command with version that was there prior to deployment, i.e. downgrade the charm. After that repeat the upgrade."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "text"
+    ]
+   },
+   "source": [
+    "## Deploy KNative and KServe charms\n",
+    "\n",
+    "KNative and KServe are new additions to Charmed Kubeflow 1.7 and need to be deployed separately as part of the upgrade:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "code"
+    ],
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# install knative and kserve\n",
+    "juju deploy knative-operator --channel 1.8/stable --trust\n",
+    "juju deploy knative-serving --config namespace=\"knative-serving\" --config istio.gateway.namespace=kubeflow --config istio.gateway.name=kubeflow-gateway --channel 1.8/stable --trust\n",
+    "juju deploy knative-eventing --config namespace=\"knative-eventing\" --channel 1.8/stable --trust\n",
+    "juju deploy kserve-controller --channel 0.10/stable --trust"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": [
+     "text"
+    ]
+   },
+   "source": [
+    "## Verify upgrade\n",
+    "\n",
+    "You can control the progress of the update by running, when all services are in `active`/`idle` state then upgrade should be finished."
```

Comment:
I didn't understand this statement.  Maybe 
> You can monitor the progress of the update by running `juju status`

?

Reply:
oh wait sorry it is for the command coming up next.

Ok maybe instead:

To monitor the progress of the update, run
```
watch -c juju status --color
```

When all services are in `active`/`idle` state, the upgrade is finished.


?